src/boot/efi/stub.c

```diff
@@ -79,20 +78,12 @@ static EFI_STATUS combine_initrd(
                 }
         }
 
-        if (credential_initrd) {
-                memcpy(p, credential_initrd, credential_initrd_size);
-                p += credential_initrd_size;
-        }
-
-        if (global_credential_initrd) {
-                memcpy(p, global_credential_initrd, global_credential_initrd_size);
-                p += global_credential_initrd_size;
-        }
-
-        if (sysext_initrd) {
-                memcpy(p, sysext_initrd, sysext_initrd_size);
-                p += sysext_initrd_size;
-        }
+        if (credential_initrd)
+                p = mempcpy(p, credential_initrd, credential_initrd_size);
+        if (global_credential_initrd)
+                p = mempcpy(p, global_credential_initrd, global_credential_initrd_size);
+        if (sysext_initrd)
+                p = mempcpy(p, sysext_initrd, sysext_initrd_size);
```

Comment:
ah, so much prettier

Reply:
Btw, I found my very first gcc bug because of this :cat: : https://gcc.gnu.org/bugzilla/show_bug.cgi?id=106868